metadata/metadata.go

```diff
@@ -174,6 +174,21 @@ func AppendToOutgoingContext(ctx context.Context, kv ...string) context.Context
 	return context.WithValue(ctx, mdOutgoingKey{}, rawMD{md: md.md, added: added})
 }
 
+// AppendToIncomingContext returns a new context with the provided kv merged
+// with any existing metadata in the context. Please refer to the documentation
+// of Pairs for a description of kv.
+func AppendToIncomingContext(ctx context.Context, kv ...string) context.Context {
```

Comment:
See comment where this function is called.

This function won't be needed.

Reply:
How important is to move adding :method POST to transport?

Reply:
It's more natural in transport. Adding it in the interceptor seems a bit hacky.

Reply:
Yeah, agreed. Deleted test, method in metadata, and code in rbac. Will add this to my transport layer PR dealing with :authority header logic at the transport layer,  or maybe I'll send it out separately. Let's take that sentence out of the scope of this PR (trust me, there's a lot more sentences to be done from A41 :D).